kamon-kafka-clients/src/main/scala/kamon/kafka/client/instrumentation/RecordProcessor.scala

```diff
@@ -0,0 +1,65 @@
+package kamon.kafka.client.instrumentation
+
+import java.time.Instant
+
+import kamon.Kamon
+import kamon.context.Context
+import kamon.kafka.client.Client
+import kamon.trace.Span
+import org.apache.kafka.clients.consumer.ConsumerRecords
+
+import scala.collection.mutable
+
+object RecordProcessor {
+
+  import scala.collection.JavaConverters._
+
+  def process[V, K](startTime: Instant, clientId: String, groupId: String, records: ConsumerRecords[K, V]): ConsumerRecords[K, V] = {
+    if (!records.isEmpty) {
+      val spanBuilder = Kamon.consumerSpanBuilder("poll", "kafka.consumer")
+        .tagMetrics("kafka.clientId", clientId)
+        .tagMetrics("kafka.groupId", groupId)
+        .tag("numRecords", records.count())
+        .tag("kafka.partitions", records.partitions().asScala.map(_.partition()).mkString(","))
+        .tag("kafka.topics", records.partitions().asScala.map(_.topic()).toSet.mkString(","))
+      val pollSpan = spanBuilder.start(startTime)
+      pollSpan.finish()
+
+      records.iterator().asScala.foreach{ record =>
+        val header = Option(record.headers.lastHeader("kamon-context"))
+
+        val sendingContext = header.map { h =>
+          ContextSerializationHelper.fromByteArray(h.value())
+        }.getOrElse(Context.Empty)
+
+          val spanBuilder = Kamon.consumerSpanBuilder("consumed-record", "kafka.consumer")
+            .tagMetrics("kafka.topic", record.topic())
+            .tagMetrics("kafka.clientId", clientId)
+            .tagMetrics("kafka.groupId", groupId)
+            .tag("kafka.partition", record.partition())
+            .tag("kafka.offset", record.offset)
```

Comment:
Tiny feature request :)
I'd love to see `record.timestamp()` and `record.timestampType()` here

Reply:
For a clearer and more precise semantic I've added a dedicated tag.